mix.exs

```diff
@@ -46,6 +46,7 @@ defmodule Twirp.MixProject do
       {:protobuf, "~> 0.5"},
       {:google_protos, "~>0.1"},
       {:finch, "~> 0.3"},
+      {:hackney, "~> 1.17"},
```

Comment:
consider making this dependency `optional: true` so that people that don't use it won't get it (well, they'd likely get it anyway from some other dep :))

```elixir
defmodule Twirp.Client.Hackney do
  @impl true
  def init() do
    if Code.ensure_loaded?(:hackney) do
      Logger.error("""
      Could not find hackney dependency.

      Please add :hackney to your dependencies:

          {:hackney, "~> 1.17"}
      """)

      raise "missing hackney dependency"
    else
      {:ok, _} = Application.ensure_all_started(:hackney)
      :ok
    end
  end

  @impl true
  def request(...)
end
```

I'm happy to prepare a PR after this lands or to this branch if you prefer!

Reply:
I just pushed a quick change for this. Can you double check me that I didn't introduce something weird? I think its correct but some of this `Code.ensure_loaded` stuff trips me up occasionally.